Packages/com.unity.inputsystem/InputSystem/InputManager.cs

```diff
@@ -354,16 +354,34 @@ public void RegisterControlLayout(string json, string name = null, bool isOverri
             }
 
             // Add it to our records.
-            var isReplacement = DoesLayoutExist(internedLayoutName);
+            var isReplacement = m_Layouts.HasLayout(internedLayoutName); 
+            if (isReplacement && isOverride)
+            {   // Do not allow a layout override to replace a "base layout" by name, but allow layout overrides
+                // to replace an existing layout override.
+                // This is required to guarantee that its a hierarchy (directed graph) rather
+                // than a cyclic graph.
+                
+                var isReplacingOverride = m_Layouts.layoutOverrideNames.Contains(internedLayoutName);
+                if (!isReplacingOverride)
+                {
+                    throw new ArgumentException($"Failed to register layout override '{internedLayoutName}'" +
+                                                $"since a layout named '{internedLayoutName}' already exist. Layout overrides must " +
+                                                $"have unique names with respect to existing layouts.");    
+                }
+            }
+
             m_Layouts.layoutStrings[internedLayoutName] = json;
             if (isOverride)
             {
-                m_Layouts.layoutOverrideNames.Add(internedLayoutName);
+                m_Layouts.layoutOverrideNames.Add(internedLayoutName); // set semantics
                 for (var i = 0; i < baseLayouts.length; ++i)
                 {
                     var baseLayoutName = baseLayouts[i];
                     m_Layouts.layoutOverrides.TryGetValue(baseLayoutName, out var overrideList);
-                    ArrayHelpers.Append(ref overrideList, internedLayoutName);
+                    if (!isReplacement)
+                        ArrayHelpers.Append(ref overrideList, internedLayoutName);
```

Comment:
I'm a bit confused, what kind of a layout will be an override but not "replacement"?

Reply:
The regular ones? As I understand this the override basically adds to the base layout as a diff while a non-override would replace an existing layout and a replacement of an existing override is a replacement and a previous override? :) At least that's how I understood @Rene-Damm when I asked about how this is supposed to work. I wonder if all of this is needed though....